addons/test_mail/tests/test_mail_gateway.py

```diff
@@ -801,6 +801,73 @@ def test_message_route_alias_multiple_new(self):
         new_simple = self.env['mail.test.gateway'].search([('name', '=', 'Test Subject')])
         self.assertEqual(len(new_simple), 1, 'message_process: a new mail.test should have been created')
 
+    @mute_logger('odoo.addons.mail.models.mail_thread', 'odoo.models')
+    def test_mail_catchall_domain_allowed_validation(self):
+        """ Check the validation of `mail.catchall.domain.allowed` system parameter"""
+        for value in (',', ',,'):
+            with self.assertRaises(exceptions.ValidationError):
+                self.env['ir.config_parameter'].set_param('mail.catchall.domain.allowed', value)
+
+        for value in ('', 'hello.com', 'hello.com,,', 'hello.com,bonjour.com', 'hello.com, bonjour.com'):
+            self.env['ir.config_parameter'].set_param('mail.catchall.domain.allowed', value)
+            value = ",".join([domain.strip() for domain in value.split(',') if domain]) if value else False
+            self.assertEqual(self.env['ir.config_parameter'].get_param('mail.catchall.domain.allowed'), value)
+
+    @mute_logger('odoo.addons.mail.models.mail_thread', 'odoo.models')
+    def test_message_route_alias_with_allowed_domains(self):
+        """ Incoming email: check that if domains are set in the
+        optional system parameter `mail.catchall.domain.allowed`,
+        only incoming emails from these domains will generate records."""
+
+        MailTestGatewayModel = self.env['mail.test.gateway']
+        MailTestContainerModel = self.env['mail.test.container']
+
+        allowed_domain = 'hello.com'
+        not_allowed_domain = 'bonjour.com'
+
+        # test@.. will cause the creation of new mail.test
+        new_alias_2 = self.env['mail.alias'].create({
+            'alias_name': 'test',
+            'alias_user_id': False,
+            'alias_model_id': self.env['ir.model']._get('mail.test.container').id,
+            'alias_contact': 'everyone',
+        })
+
+        for subject, val1, val2, domain, sys_param in [
```

Comment:
What are val1 and val2 ?

Reply:
val1 -> gateway_created ? (and make a boolean) ?
val2 -> container_created ? (and make a boolean) ?
domain -> alias2_domain ?